packages/grid_client/src/clients/tf-grid/contracts.ts

```diff
@@ -268,7 +272,7 @@ class TFContracts extends Contracts {
     const gqlClient = new Graphql(options.graphqlURL);
 
     const body = `query getConsumption($contractId: BigInt!){
-            contractBillReports(where: {contractID_eq: $contractId} , orderBy: timestamp_DESC) {
+            contractBillReports(where: {contractID_eq: $contractId} , orderBy: timestamp_DESC, limit:1) {
```

Comment:
why are we running getConsumption query twice (here) and in getConsumption function, if both queries return the discountReceived now?

Reply:
the name of the query is not related; its fine to be changed, but the functionality is different. I kept this method to be used if someone want to get only the applied discount

Reply:
I think this query is no longer necessary, what do you think @AhmedHanafy725 ?

Reply:
ok to leave it in the client